packages/engine/src/scene/systems/ShadowSystem.ts

```diff
@@ -142,62 +144,74 @@ export default async function ShadowSystem(world: World) {
   let dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, 0)
   dropShadows.matrixAutoUpdate = false
 
-  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function modifyShadowCount() {
-    world.scene.remove(dropShadows)
-    dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, shadowComponentQuery().length)
-    dropShadows.matrixAutoUpdate = false
-    dropShadows.layers.disable(ObjectLayers.Camera)
-    world.scene.add(dropShadows)
-    return null
-  })
+  const shadowComponentQuery = defineQuery([DropShadowComponent, GroupComponent])
+
+  const shadowOffset = new Vector3(0, 0.01, 0)
 
   let sceneObjects = Array.from(Engine.instance.currentWorld.objectLayerList[ObjectLayers.Camera] || [])
 
-  const CreateDropShadows = () => {
+  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function (props) {
+    if (getShadowsEnabled()) return null
```

Comment:
Any if statements must be in the effect. Having a differing number of effects in a reactor will cause bad errors in react.

Reply:
Got it, going to read up on this.